### PR TITLE
modulesync 2.5.1 and drop Puppet 4

### PR DESCRIPTION
**Pull request by bastelfreak**
modulesync 2.5.1

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,219 +1,161 @@
...
   fast_finish: true
   include:
-  - rvm: 2.1.9
-    bundler_args: --without system_tests development release
-    env: PUPPET_VERSION="~> 4.0" CHECK=test PARALLEL_TEST_PROCESSORS=12
   - rvm: 2.4.4
     bundler_args: --without system_tests development release
     env: PUPPET_VERSION="~> 5.0" CHECK=test
-  - rvm: 2.5.1
+  - rvm: 2.5.3
```
Comment (ghoneycutt) [inline .travis.yml]: seems the latest release reverted ruby to 2.5.1 according to https://puppet.com/docs/puppet/6.2/about_agent.html

File: .travis/setup.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# THIS FILE IS MANAGED BY MODULESYNC
+
+rm -f Gemfile.lock
+if [ "${PUPPET_VERSION}" = '~> 4.0' ]; then
```
Comment (ghoneycutt) [inline .travis/setup.sh]: since we're dropping version 4, no need for the conditional logic

File: Rakefile [hunk 3]
```diff
@@ -37,12 +37,27 @@ RSpec::Core::RakeTask.new(:acceptance) do |t|
   t.pattern = 'spec/acceptance'
 end
 
-desc 'Run tests metadata_lint, release_checks'
+desc 'Run tests release_checks'
 task test: [
-  :metadata_lint,
```
Comment (ghoneycutt) [inline Rakefile]: is metadata_lint called from somewhere else?

File: .travis.yml (hunk 1 above)
Comment (bastelfreak) [reply .travis.yml]: panda_tableflip

I will check why this got reverted...

File: metadata.json [hunk 4]
```diff
@@ -33,7 +33,7 @@
   "requirements": [
     {
       "name": "puppet",
-      "version_requirement": ">= 4.10.0 < 7.0.0"
+      "version_requirement": ">= 5.5.8 < 7.0.0"
```
Comment (ghoneycutt) [inline metadata.json]: 5.5.10 is the latest in the 5 series

File: .travis/setup.sh (hunk 2 above)
Comment (bastelfreak) [reply .travis/setup.sh]: I will remove it in the next modulesync version (this is currently based on a template from https://github.com/voxpupuli/modulesync_config/tree/master/moduleroot )

File: .travis.yml (hunk 1 above)
Comment (ghoneycutt) [reply .travis.yml]: lulz, also realized this is a modulesync PR, so not the right place to address these changes

File: metadata.json (hunk 4 above)
Comment (bastelfreak) [reply metadata.json]: https://voxpupuli.org/blog/2019/01/03/dropping-puppet4/ we decided on 5.5.8 back in the days. I'm not sure if we should increase the patch version or not.

File: Rakefile (hunk 3 above)
Comment (bastelfreak) [reply Rakefile]: this should be included in the release_checks task

Comment (bastelfreak) [reply Rakefile]: `validate` task calls `metadata_lint`: https://github.com/puppetlabs/puppetlabs_spec_helper/blob/db4aa99aade1b78988ab5b72078b92311bf07141/lib/puppetlabs_spec_helper/rake_tasks.rb#L211

`validate` is part of the `release_checks` task: https://github.com/puppetlabs/puppetlabs_spec_helper/blob/db4aa99aade1b78988ab5b72078b92311bf07141/lib/puppetlabs_spec_helper/rake_tasks.rb#L274

File: .travis.yml (hunk 1 above)
Comment (bastelfreak) [reply .travis.yml]: na this is fine. Feedback is always welcome. I will merge this as is (if it turns green) and afterwards update modulesync_config repo.

Comment (bastelfreak) [reply .travis.yml]: I'm not able to find a note about 2.5.1 in the changelog, nor a Jira issue. Also the latest rpm seems to include 2.5.3:

```
[root@4fb6828f1e15 /]# /opt/puppetlabs/puppet/bin/ruby -v
ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
[root@4fb6828f1e15 /]# yum info puppet-agent
Loaded plugins: fastestmirror, ovl
Loading mirror speeds from cached hostfile
 * base: mirror.fra10.de.leaseweb.net
 * extras: mirror.ratiokontakt.de
 * updates: mirror.imt-systems.com
Installed Packages
Name        : puppet-agent
Arch        : x86_64
Version     : 6.2.0
Release     : 1.el7
Size        : 93 M
Repo        : installed
From repo   : puppet6
Summary     : The Puppet Agent package contains all of the elements needed to run puppet, including ruby, facter, and hiera.
URL         : https://www.puppetlabs.com
License     : See components
Description : The Puppet Agent package contains all of the elements needed to run puppet, including ruby, facter, and hiera.
            : 
            : Contains the following components:
            : cleanup
            : cpp-hocon 0.2.0
            : cpp-pcp-client 1.6.0
            : dmidecode 2.12
            : facter 3.12.3
            : hiera 3.5.0
            : leatherman 1.5.4
            : libwhereami 0.2.2
            : module-puppetlabs-augeas_core 1.0.1
            : module-puppetlabs-cron_core 1.0.0
            : module-puppetlabs-host_core 1.0.2
            : module-puppetlabs-mount_core 1.0.2
            : module-puppetlabs-scheduled_task 1.0.0
            : module-puppetlabs-selinux_core 1.0.1
            : module-puppetlabs-sshkeys_core 1.0.2
            : module-puppetlabs-yumrepo_core 1.0.3
            : module-puppetlabs-zfs_core 1.0.1
            : module-puppetlabs-zone_core 1.0.1
            : puppet 6.2.0
            : puppet-resource_api v1.6.3-7-g2419162
            : puppet-runtime 201901130
            : pxp-agent 1.10.4
            : runtime
            : shellpath 2015-09-18
            : virt-what 1.18
            : wrapper-script

[root@4fb6828f1e15 /]#
```

I assume the docs are simply wrong. I will create an issue for that.

Comment (bastelfreak) [reply .travis.yml]: https://tickets.puppetlabs.com/browse/DOCUMENT-977

**Pull request merged**